### PR TITLE
Let's keep a CHANGELOG

### PR DESCRIPTION
**Pull request by cddr**
## What is a changelog?
A changelog is a file which contains a curated, chronologically ordered list of notable changes for each version of a project.

## Why keep a changelog?
To make it easier for users and contributors to see precisely what notable changes have been made between each release (or version) of the project.

## Who needs a changelog?
People do. Whether consumers or developers, the end users of software are human beings who care about what's in the software. When the software changes, people want to know why and how.

https://keepachangelog.com/en/1.0.0/

This PR also includes a "strawman" contributing guide to gently nudge contributions to include a changelog (as well as other guidelines that aim to describe what high-quality contributions look like)

### CODE REVIEW

File: .github/CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+### Reporting bugs
+
+When reporting bugs it's a good idea to go through the [Troubleshooting section
+of the manual][7].  Adding information like the backtrace and any sample messages
+and/or topic configurations used to the bug report makes it easier to track down bugs.
+Some steps to reproduce a bug reliably would also make a huge difference.
+
+## Pull requests
+
+* Read the [Hacking on Jackdaw][8] manual section.
```
Comment (creese) [inline .github/CONTRIBUTING]: Let's remove references to anything we haven't written yet.

File: .github/CONTRIBUTING [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+* Read [how to properly contribute to open source projects on Github][2].
+* Use a topic branch to easily amend a pull request later, if necessary.
+* Try not to update existing conventions of any code that you change.
```
Comment (creese) [inline .github/CONTRIBUTING]: Not sure what you're getting here. Is that equivalent to, "Please keep style and feature PRs separate. If you're not sure what convention to use, please ask."?

File: .github/CONTRIBUTING [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* Make sure that the unit tests are passing (`lein test`).
+* Write [good commit messages][3].
+* Mention related tickets in the commit messages (e.g. `[Fix #N] Add command ...`).
+* Update the [changelog][6].
+* [Squash related commits together][5].
+* Open a [pull request][4] that relates to *only* one subject with a clear title
+  and description in grammatically correct, complete sentences.
```
Comment (creese) [inline .github/CONTRIBUTING]: We should add instructions that all commits must be "signed off" and explain what this means. More detail on that here:
- https://about.gitlab.com/2017/11/01/gitlab-switches-to-dco-license/
- https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#sign-your-work-the-developer-s-certificate-of-origin

I started this over here so you can lift that out and close that PR:
https://github.com/FundingCircle/jackdaw/pull/56/files

File: .github/CONTRIBUTING [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+do so.
+
+## Issue reporting
+
+* Check that the issue has not already been reported.
+* Check that the issue has not already been fixed in the latest code
+  (a.k.a. `master`).
+* Open an issue with a descriptive title and a clear, concise and precise statement of the problem 
+* Mention any pertinent information about the "stack" you are using
+* Mention the version (or versions) of Jackdaw you're trying to use
```
Comment (creese) [inline .github/CONTRIBUTING]: Should we include the Kafka broker and Confluent Platform versions?

File: .github/CONTRIBUTING [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+When reporting bugs it's a good idea to go through the [Troubleshooting section
+of the manual][7].  Adding information like the backtrace and any sample messages
+and/or topic configurations used to the bug report makes it easier to track down bugs.
+Some steps to reproduce a bug reliably would also make a huge difference.
+
+## Pull requests
+
+* Read the [Hacking on Jackdaw][8] manual section.
+* Read [how to properly contribute to open source projects on Github][2].
+* Use a topic branch to easily amend a pull request later, if necessary.
```
Comment (creese) [inline .github/CONTRIBUTING]: Does "topic branch" refer to something other than what we usually mean when we say "branch"?

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## [Unreleased]
+
+### Added
+
+### Changed
+
+### Removed
+
+## [0.6.2] - [2019-03-21]
```
[the remark this answers is not in the document]
Comment (cddr) [reply CHANGELOG.md]: Yep. I updated the CHANGELOG to include the most recent activity and also updated recent release notes to add proper titles for each release.

File: .github/CONTRIBUTING (hunk 5 above)
Comment (cddr) [reply .github/CONTRIBUTING]: Just removed that line.

File: .github/CONTRIBUTING [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+of the manual][7].  Adding information like the backtrace and any sample messages
+and/or topic configurations used to the bug report makes it easier to track down bugs.
+Some steps to reproduce a bug reliably would also make a huge difference.
+
+## Pull requests
+
+* Read [how to properly contribute to open source projects on Github][2].
+* Keep style and feature PRs separate. Feel free to discuss proposals in #jackdaw[8]
+* Make sure that the unit tests are passing (`lein test`).
+* Write [good commit messages][3] and sign each commit (`git commit -s -m 'Added foo feature'`).
```
Comment (creese) [inline .github/CONTRIBUTING]: I think this should be `Add foo feature`, not `Added foo feature`.

From the link:
> Write your commit message in the imperative: "Fix bug" and not "Fixed bug"
or "Fixes bug."

File: .github/CONTRIBUTING [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+* Mention related tickets in the commit messages (e.g. `[Fix #N] Add command ...`).
+* Update the [changelog][6].
+* [Squash related commits together][5].
+* Open a [pull request][4] that relates to *only* one subject with a clear title
+  and description in grammatically correct, complete sentences.
```
Comment (creese) [inline .github/CONTRIBUTING]: `* [Sign off][10] on all commits, which certifies that you agree to the [DCO + License][9].`

File: .github/CONTRIBUTING [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+[1]: https://github.com/FundingCircle/jackdaw/issues
+[2]: http://gun.io/blog/how-to-github-fork-branch-and-pull-request
+[3]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
+[4]: https://help.github.com/articles/using-pull-requests
+[5]: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
+[6]: https://github.com/FundingCircle/jackdaw/blob/master/CHANGELOG.md
+[7]: https://github.com/FundingCircle/jackdaw/blob/master/doc/trouble-shooting.md
+[8]: https://clojurians.slack.com/messages/CEA3C7UG0/
+[9]: https://github.com/FundingCircle/jackdaw/tree/master/doc/DCO_+_LICENSE
```
Comment (creese) [inline .github/CONTRIBUTING]: `[10]: https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#sign-your-work-the-developer-s-certificate-of-origin`

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+# Changelog
+
+## [Unrelaesed]
```
Comment (creese) [inline CHANGELOG.md]: Is this section up-to-date.

File: .github/CONTRIBUTING [hunk 11]
```diff
@@ -0,0 +1,55 @@
+# Contributing to Jackdaw
```
Comment (creese) [inline .github/CONTRIBUTING]: Why is this under `.github/` and not at the project root as `CONTRIBUTING.md`?

**Pull request merged**